python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -317,21 +319,20 @@ def getMetadata(self):
             self._optionallyUploadLocalFile()
 
         # eval again here as datatime not included in dirac_ganga_server
-        r = execute('getMetadata("%s")' % self.lfn)
+
         try:
-            ret = eval(r)
+            ret = execute('getMetadata("%s")' % self.lfn)
+        except GangaDiracException as err:
+            raise
+        try:
+            if self.guid != ret['Successful'][self.lfn]['GUID']:
+                self.guid = ret['Successful'][self.lfn]['GUID']
         except:
-            ret = r
-        if isinstance(ret, dict) and ret.get('OK', False) and self.lfn in ret.get('Value', {'Successful': {}})['Successful']:
-            try:
-                if self.guid != ret['Value']['Successful'][self.lfn]['GUID']:
-                    self.guid = ret['Value']['Successful'][self.lfn]['GUID']
-            except:
-                pass
+            pass
```

Comment:
I suspect this is trying to access a potentially missing element in a dictionary, I'll go and add some exceptions here which is fixing bugs already present in develop